staging/src/k8s.io/apimachinery/pkg/util/strategicpatch/patch.go

```diff
@@ -58,13 +60,22 @@ type DiffOptions struct {
 	IgnoreChangesAndAdditions bool
 	// IgnoreDeletions indicates if we keep the deletions in the patch.
 	IgnoreDeletions bool
+	// ReplaceKeys will keep all the keys in the map or in the slice of maps.
```

Comment:
Would you update the description to make it clear that "only" the keys present in the map will be kept? Something like what's described in the proposal:

> We introduce a new value replaceKeys for patchStrategy. It indicates that all fields needing to be preserved must be present in the patch. And the fields that are present will be merged with live object. All the missing fields will be cleared when patching.